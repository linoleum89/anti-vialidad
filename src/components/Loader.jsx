import ReactDOM from 'react-dom';

const Loader = (props) => ReactDOM.createPortal(props.children, document.getElementById('overlay'));

export default Loader;