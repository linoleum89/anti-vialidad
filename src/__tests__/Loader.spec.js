import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import Loader from "../components/Loader";

describe("Loader Component", () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  it("should match the snapshot", () => {
    const component = TestRenderer.create(<Loader>Loading...</Loader>);
    const tree = component.toJSON();
    expect(tree).toEqual('Loading...')
    expect(tree).toMatchSnapshot();
  });
});