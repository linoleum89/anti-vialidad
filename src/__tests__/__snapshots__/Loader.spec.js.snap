// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should match the snapshot 1`] = `"Loading..."`;
